accord-core/src/main/java/accord/impl/SimpleProgressLog.java

```diff
@@ -358,10 +358,10 @@ void run(SafeCommandStore safeStore, SafeCommand safeCommand)
                             if (fail == null && blockedUntil.isSatisfiedBy(success))
                             {
                                 Command test = safeStore0.ifInitialised(txnId).current();
-                                Invariants.checkState(test.has(success));
+                                Invariants.checkState(test.has(success), "Command %s was expected to have known %s, but had %s", test, success, test.known());
                                 record(success);
                             }
-                        });
+                        }).begin(commandStore.agent());
```

Comment:
the current store just happens to return a `AsyncResult`, but the API does not require that, so without this `begin` this logic may never execute...

Reply:
`execute` is not a great name for a method that returns a chain that isn't started. Especially when a very similar concept is used for `Executors` and execute implies that execution is queued/begins.

Reply:
`ExecutorService` uses `Future` and `AsyncChain` is a lazy future.... I agree it's a confusing name... what also makes it hard is that the impl tend not to be lazy and to return a `AsyncResult` instead!  If we change the API to be `AsyncResult` then we are arguing eager execution rater than lazy execution...

Reply:
I just checked, InMemoryStore is eager, C* is lazy!  Got to love it!

Here is C* code

```
@Override
    public <T> AsyncChain<T> submit(Callable<T> task)
    {
        return AsyncChains.ofCallable(executor, task);
    }
...
public static <V> AsyncChain<V> ofCallable(Executor executor, Callable<V> callable)
    {
        return new Head<V>()
        {
            @Override
            protected void start(BiConsumer<? super V, Throwable> callback)
            {
                try
                {
                    executor.execute(encapsulate(callable, callback));
                }
                catch (Throwable t)
                {
                    callback.accept(null, t);
                }
            }
        };
    }
```

So, if you forget to call `begin` then this is a no-op...